presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractConnectionProperty.java

```diff
@@ -69,6 +71,18 @@ public String getKey()
         return defaultValue;
     }
 
+    @Override
+    public void setIsProvided(Properties properties)
+    {
+        isProvided = properties.containsKey(key);
```

Comment:
I don't like this field. It express the state of independent object  (`Properties` here in this case) and not actual property description. I think instead you should have local variable in the context of usage.